### PR TITLE
Use must-gather in dev-scripts for log gathering

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: must_gather.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -xe
+
+source logging.sh
+source common.sh
+
+MUST_GATHER_PATH=${MUST_GATHER_PATH:-$LOGDIR/must-gather}
```
Comment (russellb) [inline must_gather.sh]: maybe the cluster name should be in this path, too?  For me this is just `logs/must-gather/`.

**Comment by russellb** [comment]
other than potentially changing the path, this seemed to work for me

File: must_gather.sh (hunk 1 above)
Comment (stbenjam) [reply must_gather.sh]: Good point.  It probably makes sense to do that for all logs, right? I changed logging.sh to use the cluster name

Comment (stbenjam) [reply must_gather.sh]: Actually, that's problematic, since this is always sources before common.sh.

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1659/

**Comment by hardys** [comment]
Looks good but running `make gather` times out for me locally - any tips for how to debug and get more info about what the must-gather is doing?

```
+(./must_gather.sh:19): main(): oc --insecure-skip-tls-verify adm must-gather --image=virthost.ostest.test.metalkube.org:5000/localimages/local-release-image:4.5.0-0.ci-2020-03-19-134336-must-gather --dest-dir ./logs/ostest/must-gather
error: gather never finished for pod must-gather-s6dhq: timed out waiting for the condition
make: *** [Makefile:24: gather] Error 1

real	2m1.295s
user	0m0.275s
sys	0m0.108s
```

Also do we want to run this in CI to gather more debug info?

**Comment by stbenjam** [comment]
You can look at the pod must-gather created and see if the logs show anything. Was the control plane up?

**Comment by hardys** [comment]
> You can look at the pod must-gather created and see if the logs show anything. Was the control plane up?

Yeah controlplane is up, looks like a tls error:

```
[shardy@master-1 dev-scripts]$ oc get pods -A | grep gather
openshift-must-gather-g7fml                             must-gather-99ksq                                                 0/1     Init:0/1            0          1s
[shardy@master-1 dev-scripts]$ oc logs must-gather-99ksq -n openshift-must-gather-g7fml
Error from server: Get https://[fd2e:6f44:5dd8:c956::18]:10250/containerLogs/openshift-must-gather-g7fml/must-gather-99ksq/copy: remote error: tls: internal error

[shardy@master-1 dev-scripts]$ oc get node
NAME                                 STATUS   ROLES    AGE   VERSION
master-0.ostest.test.metalkube.org   Ready    master   22h   v1.17.1
master-1.ostest.test.metalkube.org   Ready    master   22h   v1.17.1
master-2.ostest.test.metalkube.org   Ready    master   22h   v1.17.1
worker-0.ostest.test.metalkube.org   Ready    worker   22h   v1.17.1
worker-1.ostest.test.metalkube.org   Ready    worker   22h   v1.17.1

[shardy@master-1 dev-scripts]$ oc get csr | grep Pending | wc -l
175
```

This cluster has been up since yesterday so we must still have CSR approval issues (build is 4.5.0-0.ci-2020-03-19-134336)

**Comment by stbenjam** [comment]
That looks like an error gathering the logs though, if you approve them, what do the logs in the actual pod say?

**Comment by hardys** [comment]
> That looks like an error gathering the logs though, if you approve them, what do the logs in the actual pod say?

My issues were CSR related (now fixed) but I now see:

```
Events:
  Type     Reason     Age        From                                         Message
  ----     ------     ----       ----                                         -------
  Normal   Scheduled  <unknown>  default-scheduler                            Successfully assigned openshift-must-gather-v55x9/must-gather-29lcb to master-1.ostest.test.metalkube.org
  Warning  Evicted    41s        kubelet, master-1.ostest.test.metalkube.org  The node had condition: [DiskPressure].
```

Strangely df doesn't seem to indicate lack of disck space on that master, I'll try again with a fresh deployment.

**Comment by stbenjam** [comment]
So, the OPENSHIFT_RELEASE_TAG isn't the right thing to create the mirrored must-gather pull-spec, it wants the version -- TAG just happens to be the version in dev-scripts, but not openshift CI.

CI builds have some version like 0.0.1-date, have to figure out how to extract that.  I also need to file a bug against must-gather since we really shouldn't have to do this, must-gather should work in disconnected just like anything else

File: must_gather.sh [hunk 2]
```diff
@@ -18,6 +18,7 @@ if [ -n "${MIRROR_IMAGES}" ]; then
 
   OPENSHIFT_RELEASE_VERSION=$(oc adm release info --registry-config="$pullsecret_file" "$OPENSHIFT_RELEASE_IMAGE" -o json | jq -r ".config.config.Labels.\"io.openshift.release\"")
   MUST_GATHER_IMAGE="--image=${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image:${OPENSHIFT_RELEASE_VERSION}-must-gather"
+  rm -f "$pullsecret_file"
```
Comment (hardys) [inline must_gather.sh]: It would be better to remove this with a trap ref https://github.com/openshift-metal3/dev-scripts/pull/936

**Comment by hardys** [comment]
Looks good although I've not managed to get this working locally.

Minor nit re trap to remove the pullsecret file, perhaps we can fix that as a followup along with whatever is needed to make it work locally (assuming its not only my env).

I know this worked in CI and will provide useful CI debug info so lets go ahead and merge, then figure out remaining issues via a followup.

**Pull request merged**